CASMCMS-9401: IMS - deleted recipe always gets assigned require_dkms=true (#6235)
Cray-HPE/docs-csm: troubleshooting/known_issues/ims_soft_deleted_recipe_always_has_require_dkms_true.md

> # Soft deleted IMS recipe always has `require_dkms` value set to `true`

Reviewer: ```suggestion
# Soft Deleted IMS Recipe Always Has `require_dkms=true`
```

## Issue description

When IMS recipe is deleted and moves to being classified as a `deleted recipe` it gets assigned `require_dkms=true`
regardless of the `require_dkms` value of the original recipe. If recipe is restored using operation `undelete`,
`require_dkms` retains the value `true`. This can be an issue in following scenario:

> - IMS recipe attribute `require_dkms` value is set to `false`.

Reviewer: ```suggestion
When an IMS recipe is deleted and becomes a "deleted recipe", its `require_dkms` attribute is set to `true`,
regardless of its original value. This new value persists even if the recipe is later restored using an `undelete`
operation.
```

1. (`ncn-mw#`) Soft delete the recipe.

    > In the following command, replace `<RECIPE_ID>` with the value of previous command.

    ```bash
    cray ims recipes delete <RECIPE_ID>
    ```

1. (`ncn-mw#`) Get the details of `soft deleted` recipe and notice the value of `require_dkms` is set to `true`.

    > In the following command, replace `<RECIPE_ID>` with the value used in previous command.

    ```bash
    cray ims deleted recipes describe <RECIPE_ID> --format json|jq '.require_dkms'
    ```

    Example output:

    ```text
    true
    ```

1. (`ncn-mw#`) `undelete` the `soft deleted` IMS recipe.

    > In the following command, replace `<RECIPE_ID>` with the value used in previous command.

    ```bash
    cray ims deleted recipes update <RECIPE_ID> --operation undelete
    ```

1. (`ncn-mw#`) Get the details of restored recipe and notice the value of `require_dkms` is set to `true`.

    > In the following command, replace `<RECIPE_ID>` with the value used in previous command.

    ```bash
    cray ims recipes describe <RECIPE_ID> --format json|jq '.require_dkms'
    ```

    Example output:

    ```text
    true
    ```

> 

Reviewer: I don't think this section is necessary. The section above already adequately explains the problem, I think.
```suggestion
```

## Resolution

> 1. (`ncn-mw#`) In order to resolve the problem, update the value of `require_dkms` to `false`.
> 
>     > In the following command, replace `<RECIPE_ID>` with the value used in `Error identification` phase.
> 
>     ```bash
>     cray ims recipes update --require-dkms false <RECIPE_ID> --format json
>     ```
> 
>     Example output:
> 
>     ```json
>     {
>     "arch": "aarch64",
>     "created": "2024-07-18T19:47:03.600611",
>     "id": "b89827ea-d929-461f-95b6-14cba7983311",
>     "link": {
>         "etag": "e07f7f535d886fb9a61130a753a2467b",
>         "path": "s3://ims/recipes/b89827ea-d929-461f-95b6-14cba7983311/recipe.tar.gz",
>         "type": "s3"
>     },
>     "linux_distribution": "sles15",
>     "name": "cray-shasta-csm-sles15sp5-barebones-csm-1.5-aarch64",
>     "recipe_type": "kiwi-ng",
>     "require_dkms": false,
>     "template_dictionary": []
>     }
>     ```

Reviewer: For procedures with only a single step, we avoid using a list.
````suggestion
(`ncn-mw#`) The problem can be worked around by manually updating the `require_dkms` value after the deleted
has been restored.

> In the following command, replace `<RECIPE_ID>` with actual IMS recipe ID.

```bash
cray ims recipes update --require-dkms false <RECIPE_ID> --format json
```

Example output:

```json
{
  "arch": "aarch64",
  "created": "2024-07-18T19:47:03.600611",
  "id": "b89827ea-d929-461f-95b6-14cba7983311",
  "link": {
    "etag": "e07f7f535d886fb9a61130a753a2467b",
    "path": "s3://ims/recipes/b89827ea-d929-461f-95b6-14cba7983311/recipe.tar.gz",
    "type": "s3"
  },
  "linux_distribution": "sles15",
  "name": "cray-shasta-csm-sles15sp5-barebones-csm-1.5-aarch64",
  "recipe_type": "kiwi-ng",
  "require_dkms": false,
  "template_dictionary": []
}
```
````

# Soft Deleted IMS Recipe Always Has `require_dkms=true`

## Issue description

When an IMS recipe is deleted and becomes a "deleted recipe", its `require_dkms` attribute is set to `true`,
regardless of its original value. This new value persists even if the recipe is later restored using an `undelete`

> operation.

Reviewer: Should add a sentence to the end of this paragraph like 
```md
This bug is fixed in CSM x.y.z. In earlier CSM versions, the only option is to use the provided [Workaround](#workaround).
```

> 

Reviewer: ```suggestion

For more information on deleting and restoring resources in IMS, see
[Delete or Recover Deleted IMS Content](../operations/image_management/Delete_or_Recover_Deleted_IMS_Content.md).

```

> ## Resolution

Reviewer: ```suggestion
## Workaround
```

(`ncn-mw#`) The problem can be worked around by manually updating the `require_dkms` value after the deleted
has been restored.

> In the following command, replace `<RECIPE_ID>` with actual IMS recipe ID.

```bash
cray ims recipes update --require-dkms false <RECIPE_ID> --format json
```

Example output:

```json
{

>   "arch": "aarch64",

Reviewer: This output does not look like it was from CSM 1.4. I only mention it because we should verify that `arch` was a possible field in IMS recipes in CSM 1.4. I don't remember when it was added.

# Soft Deleted IMS Recipe Always Has `require_dkms=true`

## Issue description

When an IMS recipe is deleted and becomes a "deleted recipe", its `require_dkms` attribute is set to `true`,
regardless of its original value. This new value persists even if the recipe is later restored using an `undelete`
operation.

For more information on deleting and restoring resources in IMS, see

> [Delete or Recover Deleted IMS Content](../operations/image_management/Delete_or_Recover_Deleted_IMS_Content.md).

Reviewer: Probably a link to this known issue should also be added somewhere on this linked page

> [Delete or Recover Deleted IMS Content](../operations/image_management/Delete_or_Recover_Deleted_IMS_Content.md).

Reviewer: ```suggestion
[Delete or Recover Deleted IMS Content](../../operations/image_management/Delete_or_Recover_Deleted_IMS_Content.md).
```
